lib/modules/manager/kustomize/extract.ts

```diff
@@ -50,10 +51,20 @@ export function extractResource(base: string): PackageDependency | null {
     return null;
   }
 
-  const { path } = match.groups;
+  const { path, queryString } = match.groups;
+  const params = querystring.parse(queryString);
+  const refParam = Array.isArray(params.ref) ? params.ref[0] : params.ref;
+  const versionParam = Array.isArray(params.version)
+    ? params.version[0]
+    : params.version;
```

Comment:
I think we should log a warning for array case, because it's pretty sure an invalid url for kustomize.

Reply:
Just tested with Kustomize, using the following url:

```
https://github.com/argoproj/argo-cd/manifests/cluster-install?ref=v2.13.1&timeout=90s&ref=v1
```

It didn't bat an eye, and used the `v2.13.1` for fetching the resource. I'd say the behavior matches this implementation, take the first value and ignore the rest.